client/src/components/MealCard.js

```diff
@@ -0,0 +1,127 @@
+    Grid,
+} from "@material-ui/core";
+import { theme } from "../themes/theme";
+import { CartContext } from "../contexts/cart/CartContextProvider";
+import { dollarFormatter } from "../lib/utils";
+
+const useStyles = makeStyles({
+    root: {
+        maxWidth: 225,
+        borderRadius: "0px",
+        boxShadow: "0px 0px 10px 5px rgba(7,7,7,0.05)",
+        margin: theme.spacing(3),
+    },
+    small: {
+        marginTop: theme.spacing(0.5),
+        marginBottom: theme.spacing(0.5),
+        marginLeft: theme.spacing(1),
+        width: theme.spacing(4),
+        height: theme.spacing(4),
+    },
+    subtitle2: {
+        fontSize: ".3rem",
+        color: theme.palette.grey[600],
+    },
+    subtitle1: {
+        fontWeight: "bold",
+        fontSize: ".7rem",
+    },
+    subtitle: {
+        fontWeight: "bold",
+    },
+});
+
+function MealCard(props) {
+    const classes = useStyles();
+    const { chef, addToCart } = useContext(CartContext);
+    const { mealPic, title, price, chefName, chefPic, location, id, chefId } = props;
+
+    const addMeal = (e) => {
+        e.preventDefault();
+        const id = parseFloat(e.currentTarget.value);
+        if (chef && chefId !== chef) {
+            alert(
+                `Your cart currently contains meals from another chef. You can only checkout with meals from one chef`
+            );
+        } else {
+            const meal = { id, mealPic, title, price, chefName, chefId };
+            addToCart(meal, id);
+        }
+    };
+
+    return (
+        <div>
+            <Card className={classes.root}>
+                <CardMedia
+                    component="img"
+                    alt="meal1"
+                    height="150"
+                    image={mealPic}
+                    title="meal1"
+                />
+                <CardContent>
+                    <Typography gutterBottom className={classes.subtitle}>
+                        {title}
+                    </Typography>
+                    <Typography
+                        gutterBottom
+                        className={classes.subtitle1}
+                        color="secondary"
+                    >
+                        {/* assuming we save price in cents on DB */}
+                        {dollarFormatter.format(price / 100)}
+                    </Typography>
+                </CardContent>
+                <Divider />
+                <CardActions>
+                    <Grid container spacing={1}>
+                        <Grid item>
+                            <Avatar
+                                className={classes.small}
+                                alt={chefName}
+                                src={chefPic}
+                            />
+                        </Grid>
+                        <Grid item xs={12} sm container alignItems="center">
+                            <Grid item xs container direction="column" spacing={2}>
+                                <Grid item xs>
```

Comment:
does sm and xs without the ={#} do anything on Typography?

Reply:
It does to do something. I actually copied this layout from a material ui example because it fit what I wanted so I am not that great at explaining it. It utilizes the auto layout feature https://material-ui.com/components/grid/#auto-layout. I can set the number of one item and the others will follow along. The exact example I copied from is here: https://material-ui.com/components/grid/#complex-grid.

Reply:
good to know. thanks for the links!